### ISSUE TITLE
Error message for incorrect endpoint does not provide enough information to debug

### ISSUE THREAD

**Issue by zombiezen** [grpc/grpc-go#2064]
I recently experienced a rather long debug session where I had misconfigured the endpoint of a gRPC connection to a Google Cloud API.  I ultimately ended up having to monkey-patch instrumentation into gRPC to figure out what was going on.

The header-handling code bails at the first error instead of trying to get a larger picture of what's going on.

https://github.com/grpc/grpc-go/blob/0ebfac42d612337c9e7fb66cbe79fe25c6b818ca/transport/http_util.go#L237-L241

### What version of gRPC are you using?

v1.11.3

### What version of Go are you using (`go version`)?

go version go1.10.2 linux/amd64

### What operating system (Linux, Windows, …) and version?

Linux

### What did you do?

Dial a gRPC connection to a non-gRPC endpoint by accident.

### What did you expect to see?

The following information would have been helpful in the error:

- **HTTP response status code.** This helps distinguish an ACL issue from a typo'd endpoint, for instance.
- **The requested URL.** Without this, I had to trace back through my code to see how the connection was being initialized to see whether this was the issue.  If I had this information in the error message, I could have simply seen the problem by a quick inspection of the error message.
- **The problematic header's error message** (already included).

### What did you see instead?

```
rpc error: code = Internal desc = transport: received the unexpected content-type "text/html; charset=UTF-8"
```

**Comment by utrack** [comment]
Also, some (configurable) way to print the body we've received would be cool.

**Comment by ayenter** [comment]
Bump

**Comment by dfawley** [comment]
Note per the linked issue that the status code here is supposed to be `Unimplemented` instead of `Internal`.

**Comment by menghanl** [comment]
This should be fixed by #2599.

Closing. Please comment if it is still a problem.
